test/trento_web/views/v1/sap_system_view_test.exs

```diff
@@ -2,41 +2,130 @@ defmodule TrentoWeb.V1.SapSystemViewTest do
   use TrentoWeb.ConnCase, async: true
 
   import Phoenix.View
-
+  import Trento.Factory
   alias Trento.Databases.Projections.DatabaseInstanceReadModel
   alias Trento.SapSystems.Projections.SapSystemReadModel
+  alias TrentoWeb.V1.SapSystemView
+
+  describe "SapSystemView" do
+    test "should render sap_systems.json" do
+      %{id: database_id, sid: database_sid} = database = insert(:database)
+
+      database_instances = [
+        build(:database_instance, database_id: database_id)
+      ]
+
+      database_instance = Enum.at(database_instances, 0)
 
-  test "should add the system replication status to the secondary instance and should remove it from the primary one" do
-    [%{database_instances: database_instances}] =
-      render(TrentoWeb.V1.SapSystemView, "sap_systems.json", %{
-        sap_systems: [
-          %SapSystemReadModel{
+      application_instances = [build(:application_instance)]
+      application_instance = Enum.at(application_instances, 0)
+
+      sap_system =
+        build(:sap_system,
+          tags: [],
+          database_id: database_id,
+          database: database,
+          database_instances: database_instances,
+          application_instances: application_instances
+        )
+
+      expected_sap_system_json =
+        [
+          %{
+            id: sap_system.id,
+            tags: [],
+            inserted_at: nil,
+            updated_at: nil,
+            health: sap_system.health,
```

Comment:
Destruct things.
I know that it might be more code, but it is much more elegant and the recommended way.
I know that we have this in other places, but the more you do the other, the more habit you will have to do that way.